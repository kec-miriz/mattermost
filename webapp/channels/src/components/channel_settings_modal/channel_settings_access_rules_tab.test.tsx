// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import type {Channel} from '@mattermost/types/channels';

import ChannelSettingsAccessRulesTab from './channel_settings_access_rules_tab';

describe('components/channel_settings_modal/ChannelSettingsAccessRulesTab', () => {
    const baseProps = {
        channel: {
            id: 'channel_id',
            name: 'test-channel',
            display_name: 'Test Channel',
            type: 'P',
        } as Channel,
        setAreThereUnsavedChanges: jest.fn(),
        showTabSwitchError: false,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <ChannelSettingsAccessRulesTab {...baseProps}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should render access rules title and subtitle', () => {
        const wrapper = shallow(
            <ChannelSettingsAccessRulesTab {...baseProps}/>,
        );

        expect(wrapper.find('.ChannelSettingsModal__accessRulesTitle')).toHaveLength(1);
        expect(wrapper.find('.ChannelSettingsModal__accessRulesSubtitle')).toHaveLength(1);
    });
});
