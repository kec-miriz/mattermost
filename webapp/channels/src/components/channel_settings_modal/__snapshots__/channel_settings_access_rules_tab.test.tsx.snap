// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/channel_settings_modal/ChannelSettingsAccessRulesTab should match snapshot 1`] = `
<div
  className="ChannelSettingsModal__accessRulesTab"
>
  <div
    className="ChannelSettingsModal__accessRulesHeader"
  >
    <h3
      className="ChannelSettingsModal__accessRulesTitle"
    >
      Access Rules
    </h3>
    <p
      className="ChannelSettingsModal__accessRulesSubtitle"
    >
      Select user attributes and values as rules to restrict channel membership
    </p>
  </div>
  <p
    className="ChannelSettingsModal__accessRulesDescription"
  >
    Select attributes and values that users must match in addition to access this channel. All selected attributes are required.
  </p>
</div>
`;
